src/Service/FinalizarVendaService.php

```diff
@@ -74,7 +74,11 @@ public function execute(int $clienteId, array $produtos): Carrinho
         }
 
         $carrinho->setStatus(StatusEnum::aguardandoPagamento);
+        $carrinho->setValorTotal(
+            $carrinho->getItems()->reduce(function ($carry, $item) {
+                return $carry + $item->getValor();
+            }, 0),
```

Comment:
da para fazer assim também, `arrow function`
acc = accumulator

```suggestion
            $carrinho->getItems()->reduce(fn ($acc, $item) => $acc + $item->getValor(), 0),
```